app/cdap/components/ConnectionList/index.tsx

```diff
@@ -0,0 +1,268 @@
+import { fetchConnectors } from 'components/Connections/Create/reducer';
+import { GCSIcon } from 'components/ConnectionList/iconStore';
+import * as React from 'react';
+import { useState } from 'react';
+import { useLocation, useParams } from 'react-router';
+import SubHeader from './Components/SubHeader';
+import ConnectionsTabs from './Components/ConnectionTabs';
+import { useStyles } from './styles';
+import If from 'components/shared/If';
+import LoadingSVG from 'components/shared/LoadingSVG';
+import ErrorSnackbar from 'components/SnackbarComponent';
+
+const SelectDatasetWrapper = styled(Box)({
+  overflowX: 'scroll',
+  display: 'flex',
+  borderTop: '1px solid #E0E0E0;',
+
+  height: '100%',
+  '& > :first-child': {
+    minWidth: '280px',
+  },
+  '& > :not(:first-child)': {
+    minWidth: '300px',
+  },
+});
+
+const DatasetWrapper = () => {
+  const { connectorType } = useParams() as any;
+
+  const classes = useStyles();
+  const loc = useLocation();
+  const queryParams = new URLSearchParams(loc.search);
+  const pathFromUrl = queryParams.get('path') || '/';
+  const [loading, setLoading] = useState(true);
+  const [isErrorOnNoWorkspace, setIsErrorOnNoWorkSpace] = useState<boolean>(false);
+
+  const toggleLoader = (value: boolean, isError?: boolean) => {
+    setLoading(value);
+    if (isError) {
+      setIsErrorOnNoWorkSpace(isError);
+    }
+  };
+  let connectionId;
+  const [dataForTabs, setDataForTabs] = useState([
+    {
+      data: [],
+      showTabs: true,
+      selectedTab: null,
+      isSearching: false,
+    },
+  ]);
+
+  const getConnectionsTabData = async () => {
+    let connectorTypes = await fetchConnectors();
+    let allConnectionsTotalLength = 0;
+
+    const categorizedConnections = await getCategorizedConnections();
+    connectorTypes = connectorTypes.filter((conn): any => {
+      return [conn.name];
+    });
+
+    connectorTypes = connectorTypes.map((connectorType): any => {
+      const connections = categorizedConnections.get(connectorType.name) || [];
+      allConnectionsTotalLength = allConnectionsTotalLength + connections.length;
+
+      return {
+        ...connectorType,
+        count: connections.length,
+        SVG: <GCSIcon />,
+      };
+    });
+    const firstLevelData = connectorTypes.filter((each) => {
+      if (each.count > 0) {
+        return {
+          name: each.name,
+          count: each.count,
+        };
+      }
+    });
+    setLoading(false);
+    setDataForTabs((prev): any => {
+      const tempData = [...prev];
+      tempData[0].data = firstLevelData;
+      return tempData;
+    });
+  };
+
+  const getCategorizedConnectionsforSelectedTab = async (selectedValue: string, index: number) => {
+    const categorizedConnections = await getCategorizedConnections();
+    const connections = categorizedConnections.get(selectedValue) || [];
+
+    setDataForTabs((prev): any => {
```

Comment:
can we let typescript infer this type? was it complaining here?

Reply:
Done